Tasks/AzureIoTEdgeV2/deployimage.ts

```diff
@@ -43,18 +43,42 @@ class azureclitask {
 
       this.loginAzure();
 
-      tl.debug('OS release:' + os.release());
-
+      tl.debug('OS release:' + os.release());      
       let showIotExtensionCommand = ["extension", "show", "--name", "azure-iot"];
       let result = tl.execSync('az', showIotExtensionCommand, Constants.execSyncSilentOption);
       if (result.code !== 0) { // The extension is not installed
+        let checkAzureIoTVersionExtensionCommand = ["--version"]; // temporary solution until the paho-mqtt situation goes away 
         let installFixedIotExtensionCommand = ["extension", "add", "--source", Constants.azureCliIotExtensionDefaultSource, "-y", "--debug"];
         let installLatestIotExtensionCommand = ["extension", "add", "--name", "azure-iot", "--debug"];
         try {
+          let outputStream: EchoStream = new EchoStream();
+          let execOptions: IExecOptions = {
+            errStream: outputStream as stream.Writable
+          } as IExecOptions;
+          
+          // check azcli version
+          let viewAzVersionResult = await tl.exec('az', checkAzureIoTVersionExtensionCommand, execOptions);
+          if(viewAzVersionResult !== 0)
+          {
+            throw new Error(`View Az Version Error: ${outputStream.content}`);
+          }
+          
+          // upgrade setuptools in private azcli environment, fix for windows only
```

Comment:
>fix for windows only [](start = 62, length = 20)

is the setuptools upgrade required because the tools cache has another version? If so, then I expect that the az cli team has a tracking item to update the toolscache on win2016 images, otherwise all scenarios in this environment will require this workaround. Alternatively, can the az cli installation handle this specific requirement as part of its install?

Reply:
The setuptools command is because az-cli runs in its own private python environment and thus not relevant to the version installed in the toolscache.